From 9a04e733fac1874d6b99a9906f5e8b3b4b567f0e Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Sun, 30 Jun 2013 02:41:43 +0200
Subject: [PATCH 2/2] Fix printing version information in berdecode

The crashes in the printing logic has been reported by The Mayhem Team
at http://www.forallsecure.com/bug-reports/6f5f79ad3f80cffe072b7133a3b8224d1b18d2b5
---
 tbl-tools/berdecode/berdecode.c |   20 +++-----------------
 1 file changed, 3 insertions(+), 17 deletions(-)

diff --git a/tbl-tools/berdecode/berdecode.c b/tbl-tools/berdecode/berdecode.c
index 012ef86..62b0001 100644
--- a/tbl-tools/berdecode/berdecode.c
+++ b/tbl-tools/berdecode/berdecode.c
@@ -1,6 +1,7 @@
 #include "tbl-gen.h"
 #include "sbuf.h"
 #include "tbl-dbg.h"
+#include "version.h"
 
 #if TIME_WITH_SYS_TIME
 # include <sys/time.h>
@@ -189,27 +190,12 @@ int main (int argc, char* argv[])
 	}
     if (!files)
 	{
-	char* revision = "$Revision: 1.3 $";
-	char* date = "$Date: 1997/09/16 15:05:10 $";
-	char* p;
-	p = strchr(revision,' ');
-	if (p)
-	    revision = p+1;
-	p = strchr(revision,' ');
-	if (p)
-	    *p = '\0';
-	p = strchr(date,' ');
-	if (p)
-	    date = p+1;
-	p = strchr(date,' ');
-	if (p)
-	    *p = '\0';
 	fprintf(stderr,"Usage: %s [-s|-strip <bytestostrip>] [-d|-debug]\n",
 		progname);
-	fprintf(stderr,"        [-T|-table <tablefilename] {<filename>|-}\n");
+	fprintf(stderr,"        [-T|-table <tablefilename>] {<filename>|-}\n");
 	fprintf(stderr,"\n");
 	fprintf(stderr,"This is berdecode, revision %s as of %s.\n",
-		revision,date);
+		VERSION, RELDATE);
 	fprintf(stderr,"This program reads a binary ASN.1 grammar file generated by snacc -T\n");
 	fprintf(stderr,"and uses it to decode ASN.1 BER encoded data from files or stdin.\n");
 	fprintf(stderr,"\n");
-- 
1.7.10.4

